src/btree/bt_read.c

```diff
@@ -862,14 +867,13 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
 
 		/*
 		 * If stalling and this thread is allowed to do eviction work,
-		 * check if the cache needs help. If we do work for the cache,
-		 * substitute that for a sleep.
+		 * check if the cache needs help evicting clean pages (don't
+		 * force a read to do dirty eviction).  If we do work for the
+		 * cache, substitute that for a sleep.
 		 */
 		if (!LF_ISSET(WT_READ_IGNORE_CACHE_SIZE)) {
 			WT_RET(__wt_cache_eviction_check(
-			    session, true,
-			    !F_ISSET(&session->txn, WT_TXN_HAS_ID),
-			    &cache_work));
```

Comment:
Some subtlety here: I don't think readers should pay to do dirty eviction while waiting for a page read.